cache-config/testing/ort-tests/t3c-check-reload_test.go

```diff
@@ -16,24 +16,44 @@ package orttest
 
 import (
 	"encoding/json"
+	"fmt"
 	"strings"
 	"testing"
 
 	"github.com/apache/trafficcontrol/cache-config/t3cutil"
 )
 
-func TestCheckReload(t *testing.T) {
-	type ChangedCfg struct {
-		ChangedFiles     string `json:"changed_files"`
-		InstalledPlugins string `json:"installed_plugins"`
-	}
+type ChangedCfg struct {
```

Comment:
Is there a reason to move `argsResults`, `ChangedCfg`, and `testCheckReload` outside `TestCheckReload`? It's always good to keep scope as small as possible, to reduce bugs. 
 
I also find it a lot easier to read when the struct is declared right where it's used, so I don't have to keep jumping back and forth. 

I also find it easier to read tests like this directly where they're called, `t.Run(testName, func(t *testing.T) { ... })` without a separate function. Since it's only called once, and I basically always want to see what it's doing when reading the outer test, and basically never want to read the function outside that context.

But I realize those things are relative, if other people feel strongly against them.

Reply:
There is a reason, though whether it's "good" depends on whether or not you think moving the loop contents into a sub-test was a positive change. The function argument needs that type in its scope, and I think using that function as a sub-test self-contains the logic for each tested scenario nicely. So I would call that a good reason, because like I said above I'm generally of the opinion that functions should have names. Separates concerns more, IMO even though it does nominally increase scope for things not used in those outer scopes. I don't think the content of either function is necessary implementation detail to understand the purpose of the other.